libsatadrivers/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright 2019, DornerWorks
+# Copyright 2018, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+cmake_minimum_required(VERSION 3.7.2)
+
+project(libsatadrivers C)
+
+set(configure_string "")
+
+config_option(LibSatadrivers LIB_SATA "Support library for x86 SATA" DEFAULT OFF)
+
+config_option(
+    LibSatadriversAHCIEnable
```

Comment:
A more CMake-friendly way of implementing this option would be to create two libraries: satadrivers-ahci and satadrivers-ide.  Then the application can specify which library to link, but the mode no longer has to be a global setting which would disallow one component from having an achi based driver and another having ide.

Reply:
I think I could refactor parts of the library to allow both the AHCI and IDE drivers to co-exist. The users can then just pick whichever one to use depending on their needs.